proposed/2022/Package-Source-Mapping-VS-Options.md

```diff
@@ -0,0 +1,145 @@
+# Package Source Mapping in Visual Studio Options Dialog
+
+* Start Date: 06-10-2022
+* Authors: Ella McNally
+* Issue: https://github.com/NuGet/Home/issues/11363
+
+## Summary
+
+Currently there is no support for package source mapping in the Visual Studio options dialog. I will be creating a feature where users can pin default sources and choose a specific source(s) for certain package source mappings from sources already provided in package sources page. This feature is a portion of what was discussed in a previous spec. https://github.com/NuGet/Home/blob/dev/implemented/2021/PackageSourceMapping.md
+
+## Motivation 
+
+Adding support for package source mapping in the VS options dialog will allow user to pin sources and add/remove package source mappings with more ease.
+
+## Explanation
+
+### Functional Explanation
+
+#### Package Source Mappings Page 
+
+When the user opens the package source mapping page in the options dialog, they will see a list of all the package source mappings they have already configured and the filepath to the solution level config file. There is a hierarchy of NuGet configs levels: machine, user, and solution. Package source mappings will be read from all of the levels, but only written to the solution level config. This page will have three buttons: _Add_ , _Remove_, and _Clear_.

```

Comment:
I think this could probably use a clarification. 
What if there's no solution level config? We don't really recognize a solution level config, it's just another config for us, so they may prove to be challenging to implement. 

Additionally, it seems like we'll only display the mappings if they are enabled, so maybe a lot of concerns about where we write mappings is moot.